pkg/reconciler/namespace/namespace.go

```diff
@@ -326,14 +342,23 @@ func clusterLabelPatchBytes(val string) []byte {
 func (c *Controller) observeCluster(ctx context.Context, cl *clusterv1alpha1.Cluster) error {
 	klog.Infof("Observing Cluster %s", cl.Name)
 
-	if conditions.IsTrue(cl, clusterv1alpha1.ClusterReadyCondition) {
+	evicted := cl.Spec.EvictAfter != nil && cl.Spec.EvictAfter.Time.Before(now())
+
+	// If cluster is Ready and not unschedulable or evicting, revisit any
+	// unscheduled namespaces.
+	if conditions.IsTrue(cl, clusterv1alpha1.ClusterReadyCondition) && !cl.Spec.Unschedulable && evicted {
 		// Revisit any unscheduled namespaces
 		var errs []error
 		errs = append(errs, c.enqueueNamespaces(ctx, unscheduledSelector))
 		errs = append(errs, c.enqueueNamespaces(ctx, scheduledToNothingSelector))
 		return errors.NewAggregate(errs)
 	}
 
+	if cl.Spec.EvictAfter != nil && cl.Spec.EvictAfter.After(now()) {
+		dur := now().Sub(cl.Spec.EvictAfter.Time)
```

Comment:
if now is before eviction time, now-evictAfter is negative. Is that intentional?